packages/common/src/llm/openai.ts

```diff
@@ -259,16 +274,30 @@ function mapToStopReason(
 
 function mapToToolCalls(
   openAIToolCalls: ChatCompletionMessageToolCall[] | undefined,
-  logger: IntegrationLogger
+  logger: IntegrationLogger,
+  provider: string
 ): ToolCall[] | undefined {
   return openAIToolCalls?.reduce((toolCalls, openAIToolCall) => {
     if (openAIToolCall.type === 'function') {
+      let toolCallArguments: ToolCall['function']['arguments']
+
+      try {
+        toolCallArguments = JSON.parse(openAIToolCall.function.arguments)
+      } catch (err) {
+        logger
+          .forBot()
+          .warn(
+            `Received invalid JSON from "${provider}" LLM provider for arguments in a tool call of function "${openAIToolCall.function.name}", a \`null\` value for arguments will be passed instead - JSON parser error: ${err} / Invalid JSON received: ${openAIToolCall.function.arguments}`
```

Comment:
:eyes: **nit (non-blocking):** err.message, the stack trace might make the message harder to read

Reply:
I'll refactor this to only include specific properties of the error.